apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -1195,6 +1197,10 @@ namespace MWRender
 
         mUniformNear->set(mNearClip);
         mUniformFar->set(mViewDistance);
+
+        // Since our fog is not radial yet, we should take FOV in account, otherwise terrain near viewing distance may disappear.
+        float distanceMult = std::cos(osg::DegreesToRadians(mFieldOfView)/2.f);
+        mTerrain->setViewDistance(mViewDistance * (distanceMult ? 1.f/distanceMult : 1.f));
```

Comment:
I think it should still make sense to put an upper limit on the multiplier, something like 2 (110 degrees FOV) or 3 (140 degrees FOV). Or it'll be 114 if 179 degree FOV is used.